cpp/include/libxtreemfs/file_handle_implementation.h

```diff
@@ -77,6 +165,12 @@ class XCapManager :
   /** Get the file id from the capability. */
   uint64_t GetFileId();
 
+  /** Returns the list of old expire times. */
+  std::list< ::google::protobuf::uint64> GetOldExpireTimes();
+
+  /** Returns the mutex related to list of old expire times. */
+  boost::mutex& GetOldExpireTimesMutex();
```

Comment:
I think returning a reference to a mutex is not a good style. You should encapsulate this and provide methods to acquire and release a lock.